BSPConversionLib/Source/EntityConverter.cs

```diff
@@ -551,6 +551,50 @@ private void GiveWeaponOnStartTouch(Entity trigger, Entity weaponEnt)
 				fireOnce = -1
 			};
 			trigger.connections.Add(connection);
+
+			if (weaponEnt["count"] == "-1")
+				return;
+
+			if (!weaponEnt.TryGetValue("count", out _) || weaponEnt["count"] == "0") // Every quake weapon has a default ammo count when none is specified
+				SetDefaultAmmoCount(weaponEnt, weaponEnt.ClassName);
+			
+			var ammoType = GetWeaponAmmoType(weaponEnt.ClassName);
+			if (string.IsNullOrEmpty(ammoType))
+				return;
+			else
+			{
+				var connection2 = new Entity.EntityConnection()
+				{
+					name = "OnStartTouch",
+					target = "!activator",
+					action = ammoType,
+					param = weaponEnt["count"],
+					delay = 0f,
+					fireOnce = -1
+				};
+				trigger.connections.Add(connection2);
+			}
+
+		}
+		private string SetDefaultAmmoCount(Entity weaponEnt, string weaponName)
```

Comment:
Rename this to `GetDefaultAmmoCount` and only return the ammo count. Then change line 559 like so:
`weaponEnt["count"] = GetDefaultAmmoCount(weaponEnt.Classname);`